### PR TITLE
docker

### PR DESCRIPTION
**Pull request by jthandy**

### CODE REVIEW

**Comment by jthandy** [comment]
maybe you could play with this and see if you can get it working?

**Comment by drewbanin** [comment]
This is tight! I'm not too familiar with docker -- what's the use case here? Is it so this works on windows/etc?

Also: a handy function to add to your bashrc is:
```bash
function dbt() {
	docker run -v ${PWD}:/dbt -v ~/.dbt:/root/.dbt jthandy/dbt /bin/bash -c "dbt ${1}"
}
```

Feel free to update the docs accordingly!

**Comment by jthandy** [comment]
@drewbanin updated readme. should be good to go. 👍 ?

**Comment by jthandy** [comment]
Also: as far as use case. We now have a couple of different tools that require different things from an environment perspective. The goal of this is simply to isolate the things that dbt requires in order to run into their own container so that they can be abstracted from the rest of the things we need. I'm also going through this process with the jupyter dependencies we have, because I attempted to do a code review for @3mei last week and couldn't do it because of problems like this. At this point, dbt and jupyter can both be run from containers and don't have any particular requirements for the host machine other than a working docker install.

**Comment by 3mei** [comment]
@jthandy Glad you are finally liking Docker! ;) I can't pull the image due to Docker hub errors (`Error while pulling image: Get https://index.docker.io/v1/repositories/jthandy/dbt/images: dial tcp: lookup index.docker.io on 75.75.76.76:53: server misbehaving`), but the Docker file looks very similar to the one I used to demonstrate dbt on Docker to you and Sam. @drewbanin The use case is that we don't have to worry about different operating systems/dependencies because everyone will be running a Linux kernel and using the image we provide.

**Comment by jthandy** [comment]
@3mei i have no idea what is going on there--sounds like a docker problem though...

**Comment by 3mei** [comment]
@jthandy @drewbanin Minor fix for when the user's folder name has white spaces in it and `${PWD}` doesn't work:

```
function dbt() {
    docker run -v "${PWD}":/dbt -v ~/.dbt:/root/.dbt jthandy/dbt /bin/bash -c "dbt ${1}"
}
```

Otherwise works well for me. The hub image issue seems to have been resolved.

File: README.md [hunk 1]
```diff
@@ -62,6 +62,20 @@ user:                         # you can have multiple profiles for different pro
...
 
+#### docker
+
+an alternate means of using dbt is with the docker image jthandy/dbt. if you already have docker installed on your system, run with:
+```docker run -v ${PWD}:/dbt -v ~/.dbt:/root/.dbt jthandy/dbt /bin/bash -c "[type your command here]"
+this can be run from any dbt project directory. it relies on the same configuration setup outlined above.
+
+on linux and osx hosts, running this can be streamlined by including the following function in your bash_profile:
+```
+function dbt() {
```
Comment (3mei) [inline README.md]:
```
function dbt() {
    docker run -v ${PWD}:/dbt -v ~/.dbt:/root/.dbt jthandy/dbt /bin/bash -c "dbt ${1}"
}
```

File: README.md [hunk 2]
```diff
@@ -62,6 +62,20 @@ user:                         # you can have multiple profiles for different pro
...
+    docker run -v ${PWD}:/dbt -v ~/.dbt:/root/.dbt jthandy/dbt /bin/bash -c "dbt ${1}"
```
Comment (3mei) [inline README.md]: Same here.

File: README.md (hunk 1 above)
Comment (jthandy) [inline README.md]: @3mei I don't think this actually works... i tested it out on a folder with spaces and got an error. I think you generally have to escape spaces as opposed to quote them in bash, but I'm not an expert on that...regardless it didn't work. 

I'm not sure that we should spend a lot of time troubleshooting a helper script though--would rather just push this.

**Comment by jthandy** [comment]
@drewbanin just waiting on you.

**Comment by drewbanin** [comment]
Approved

**Pull request merged**